scipy/stats/_continuous_distns.py

```diff
@@ -10481,12 +10481,20 @@ class tukeylambda_gen(rv_continuous):
     %(example)s
 
     """
+    _support_mask = rv_continuous._open_support_mask
+
     def _argcheck(self, lam):
         return np.isfinite(lam)
 
     def _shape_info(self):
         return [_ShapeInfo("lam", False, (-np.inf, np.inf), (False, False))]
 
+    def _get_support(self, lam):
+        b = _lazywhere(lam > 0, (lam,),
+                       f=lambda lam: 1/lam,
+                       f2=lambda lam: np.inf)
```

Comment:
Just to show that I'm actually thinking about this - not that there's any real motivation to change this - it would be slightly _simpler_ as:
```suggestion
        b = _lazywhere(lam <= 0, (lam,), np.inf, f2=1/lam)
```
right?
And `_argcheck` should already be taking care of when `lam` is invalid, but I suppose only this way would do the right thing if `lam = np.nan`.

Reply:
That's not quite right, but you are right that it can be simplified: replace `f2=lambda lam: np.inf` with `fillvalue=np.inf`.  I'll push that change.

Reply:
Oops, should have checked the signature. OK, I'll merge after that.